test/distributed/org/apache/cassandra/distributed/test/sai/StrictFilteringTest.java

```diff
@@ -68,9 +65,7 @@ public void shouldRejectNonStrictIN()
 
         String select = withKeyspace("SELECT * FROM %s.reject_in WHERE a = 1 AND b IN (2, 3) ALLOW FILTERING");
 
-        // This should fail, as strict filtering is not allowed:
-        Assertions.assertThatThrownBy(() -> CLUSTER.coordinator(1).execute(select, ConsistencyLevel.ALL))
-                  .hasMessageContaining(String.format(StorageAttachedIndexQueryPlan.UNSUPPORTED_NON_STRICT_OPERATOR, Operator.IN));
+        assertRows(CLUSTER.coordinator(1).execute(select, ConsistencyLevel.ALL), row(0,1,2));
```

Comment:
Two notes:

1.) We can probably rename the test something like `shouldPostFilterNonStrictIN`

2.) We don't really need the index on `b` now.